File: fluentcore/src/main/java/com/mcreater/fxfluent/xaml/tags/DynamicColorContentTag.java
package com.mcreater.fxfluent.xaml.tags;

import javafx.scene.paint.Color;

import java.util.function.Supplier;

public class DynamicColorContentTag extends ColorContentTag {
    private final Supplier<Color> consumer;
    private final String key;
    public DynamicColorContentTag(String key, Supplier<Color> consumer) {
        this.consumer = consumer;
        this.key = key;
    }
    public Color toObject() {
        return consumer.get();
    }

    public String getKey() {
        return key;
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/xaml/tags/DynamicColorContentTag.kt
package com.mcreater.fxfluent.xaml.tags

import javafx.scene.paint.Color
import java.util.function.Supplier


class DynamicColorContentTag(private val key: String, private val consumer: Supplier<Color>) : ColorContentTag() {
    override fun toObject(): Color {
        return consumer.get()
    }

    override fun getKey(): String {
        return key
    }
}

